src/phylum/logger.py

```diff
@@ -0,0 +1,236 @@
+"""Configure the logging features for the package."""
+import inspect
+import logging
+import sys
+from types import FunctionType, MethodType
+from typing import Any, Callable, Dict, Optional
+
+from rich.logging import RichHandler
+from rich.progress import Progress, SpinnerColumn, TextColumn, TimeElapsedColumn
+import rich.traceback
+
+from phylum import PKG_NAME
+from phylum.console import console
+
+LOG = logging.getLogger(PKG_NAME)
+
+# This is a custom logging level, defined relative to existing logging.DEBUG level to make type checkers happy.
+LOGGING_TRACE_LEVEL = logging.DEBUG - 5
```

Comment:
This strikes me as extremely odd. Could you elaborate on why this is the way it is?

Reply:
I tried again just now and the bit about the type checkers is not true...not sure what I saw the first time to think that. [This is how Python logging levels are defined](https://devdocs.io/python~3.8/library/logging#levels). Setting a value relative to a known value was done to have less of a "magic number" in the code. It can be changed though now that the type checking bit is cleared up.